### PR TITLE
Changes for daily latest for control index

### PR DESCRIPTION
**Pull request by SahithiMy**
Signed-off-by: SahithiMy <sahithi.mylangam@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Added a function to change the daily latest flag for 24 hours for new control index.

### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/STALWART/boards/361?modal=detail&selectedIssue=STALWART-185

### :+1: Definition of Done
Changed the daily latest for new control index structure

### :athletic_shoe: How to Build and Test the Change
rebuild components/compliance-service/
chef_load_compliance_scans
### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [X] Tests added/updated? (all new code needs new tests)
- [X] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 1]
```diff
@@ -993,3 +993,33 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
...
+		Must(termQueryThisControl)
+
+	script := elastic.NewScript(`
+		 def targets = ctx._source.nodes.findAll(node -> node.node_uuid == params.node_uuid);
+		 for(node in targets) { 
+			 if(node.daily_latest==true) {
+				 node.daily_latest = false
+			 }
+		 }
+		 if (ctx._source.daily_latest==true) {
```
Comment (YashviJain01) [inline components/compliance-service/ingest/ingestic/ingestic.go:1010]: We can remove this if statement.

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 2]
```diff
@@ -993,3 +993,33 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
...
+			 ctx._source.daily_latest = false;
+		 }`).Param("node_uuid", nodeId)
+	esIndexes, err := relaxting.IndexDates(relaxting.CompDailyControlIndexPrefix, time.Now().Add(-24*time.Hour).Format(time.RFC3339), time.Now().Format(time.RFC3339))
```
Comment (YashviJain01) [inline components/compliance-service/ingest/ingestic/ingestic.go:1013]: Some compliation error

Comment (YashviJain01) [inline components/compliance-service/ingest/ingestic/ingestic.go:1013]: We just need today's index, so no need to call method Index Dates

**Review by YashviJain01 — APPROVED** [review]

**Review by kalroy — APPROVED** [review]

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 3]
```diff
@@ -993,3 +993,30 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
...
+	// Script to find the nodes and making daily_latest as false
+	script := elastic.NewScript(`
+		 def targets = ctx._source.nodes.findAll(node -> node.node_uuid == params.node_uuid);
+		 for(node in targets) { 
+			 if(node.daily_latest==true) {
+				 node.daily_latest = false
+			 }
+		 }`).Param("node_uuid", nodeId)
+
+	// Updating in all the Indices
```
Comment (YashviJain01) [inline components/compliance-service/ingest/ingestic/ingestic.go:1014]: Please change the comment to : Update in current Index.

File: components/compliance-service/ingest/pipeline/publisher/compliance.go [hunk 4]
```diff
@@ -74,7 +74,6 @@ func storeCompliance(in <-chan message.Compliance, out chan<- message.Compliance
 			message.Propagate(out, &msg)
 		}
 		logrus.WithFields(logrus.Fields{"report_id": msg.Report.ReportUuid}).Debug("Published Compliance Report")
-
```
Comment (YashviJain01) [inline components/compliance-service/ingest/pipeline/publisher/compliance.go:77]: Why this change coming as a part?

**Review by YashviJain01 — APPROVED** [review]
